### ISSUE TITLE
Autorest: Create deployment from template

### ISSUE THREAD

**Issue by dtretyakov** [Azure/azure-sdk-for-java#625]
Before beta1 it was possible to pass template and parameters produced by `ObjectMapper` deserializer like that:
```java
final ObjectMapper mapper = new ObjectMapper();
final JsonNode template = mapper.readTree(templateValue);
final JsonNode parameters = mapper.readTree(parametersValue);

deployment.setProperties(new DeploymentProperties());
deployment.getProperties().setMode(DeploymentMode.INCREMENTAL);
deployment.getProperties().setTemplate(template);
deployment.getProperties().setParameters(parameters);

deployments.createOrUpdate(GROUP_ID, DEPLOYMENT_ID, deployment);
```

Currently `DeploymentsOperationsImpl.createOrUpdate` method throws `StackOverflowError` as follows:
```java
java.lang.StackOverflowError
	at java.util.HashMap$Node.<init>(HashMap.java:284)
	at java.util.HashMap.newNode(HashMap.java:1734)
	at java.util.HashMap.putVal(HashMap.java:630)
	at java.util.HashMap.put(HashMap.java:611)
	at com.google.common.reflect.TypeToken$TypeCollector.collectTypes(TypeToken.java:1154)
	...
	at com.google.common.reflect.TypeToken$TypeCollector.collectTypes(TypeToken.java:1127)
	at com.google.common.reflect.TypeToken$TypeCollector$3.collectTypes(TypeToken.java:1114)
	at com.google.common.reflect.TypeToken$ClassSet.rawTypes(TypeToken.java:664)
	at com.microsoft.rest.Validator.validate(Validator.java:59)
	at com.microsoft.rest.Validator.validate(Validator.java:94)
       ...
	at com.microsoft.rest.Validator.validate(Validator.java:94)
```

How should be passed deployment template and parameters at the moment?

**Comment by dtretyakov** [comment]
As a workaround template and parameters strings can be wrapped by POJO like that:
```java
public class Pojo {
    private final String myJson;

    Pojo(final String json) {
        myJson = json;
    }

    @JsonValue
    @JsonRawValue
    @Override
    public String toString() {
        return myJson;
    }
}
```
But what is a better way to do that? May be you can add example for template deployment.

**Comment by jianghaolu** [comment]
Hi @dtretyakov That was a known bug in beta1 similar to #820. This is fixed in snapshot version 1.0.0-20160309.002843-19.
